src/lib/EDSL.mli

```diff
@@ -4,6 +4,15 @@ type 'a t = 'a Language.t
 val fail: unit t
 (** Abort the script/command immediately. *)
 
+val with_throw:
+  ?using_signal:string -> catch:unit t -> (unit t -> unit t) -> unit t
+(** Use a UNIX signal (default ["USR1"]) to create a “jump.”
+
+    [with_throw ~catch (fun throw -> (* more_code *))]
+    executes [(* more_code *)] but if it uses [throw], the code behaves like
+    a raised exception, and the [catch] argument is executed.
+ *)
```

Comment:
Naming:

Since "throw"'s first meaning is a verb and "with" denotes possession of a thing, I find this name a bit awkward. One could use `with_exception` and label the first argument as an `exception` but that has a dissonance with the programmatic meanings of "exception". So I would ultimately recommend `with_signal` and `signal` for the first argument. A follow on change would be `using_signal` -> `signal_name`. 

Example:
I think you really need an example in this documentation to make it easier to understand what is going on.